File: examples/functions/primes.cpp
#include <iostream>
#include <cmath>

bool is_prime(int N) {

    bool prime{true};

    int max_factor = static_cast<int>(std::sqrt(N));

    for (int q = 2; q <= max_factor; ++q) {
        if (N % q == 0) {
            prime = false;
            break;
        }
    }

    return prime;
}

int main() {

    const int n_max = 100;

    for (int n = 2; n <= n_max; ++n) {
        if (is_prime(n)) {
            std::cout << n << " is prime!" << std::endl;
        }
    }

}

File: docs/source/homework/primes.py
import math


def is_prime(N):

    prime = True

    max_factor = int(math.sqrt(N))

    for q in range(2, max_factor+1):
        if N % q == 0:
            prime = False
            break

    return prime

N_MAX = 100

for n in range(2, N_MAX+1):
    if is_prime(n):
        print(f"{n} is prime!")
